src/ipc/ipc4/handler.c

```diff
@@ -516,13 +516,18 @@ static int ipc_wait_for_compound_msg(void)
 	int try_count = 30; /* timeout out is 30 x 10ms so 300ms for IPC */
 
 	while (atomic_read(&msg_data.delayed_reply)) {
+		/* preventing clock switching in idle */
+		pm_runtime_disable(CORE_HP_CLK, PLATFORM_PRIMARY_CORE_ID);
 		k_sleep(Z_TIMEOUT_MS(10));
```

Comment:
I don't know the hardware details, but naively this feels like the opposite to what one would want: when sleeping you'd want preferably to switch off any clocks, do some kind of a wait-for-interrupt operation, and instead we turn the clock on to a maximum frequency?

Reply:
Maybe I implemented it in the wrong function. But the goal here is not to turn on the highest clock but to prevent switching to the lowest one if a higher one is already in use.